### PR TITLE
Fix handling of `describeKeyspace` to catch RuntimeEcxeptions too, add a test

### LINKED ISSUE #2179: Bridge service is not handling runtime exceptions
Bridge service currently silently ignores the `RuntimeExceptions`, if they occur in the methods that are not handled with the `MessageHandlers`. Basically:

* `executeQueryWithSchema`
* `describeKeyspace`

The proper try/catch must be included, so that any runtime exception is handled. Current situation is that client side hangs, because observer on the server side is not informed on the error.

We need to catch that hang in test first. This can be easily done, mock the `Persistence` and for example throw exception on `persistence.decorateKeyspaceName`..

In addition to this, these exceptions should be logged. Also we should log possibly all the exceptions that we catch in the `io.stargate.bridge.service.ExceptionHandler`, which is not done right now.

### PR DESCRIPTION
**Pull request by tatu-at-datastax**
**What this PR does**:

Improves bridge handling so that:

* `describeKeyspace()` operation will catch all `Throwable`s and report them properly (not just gRPC fails)
* `queryWithSchema()` operation that fails internally will also report failure gracefully (add test(s))

**Which issue(s) this PR fixes**:
Fixes #2179

**Checklist**
- [ ] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Comment by tatu-at-datastax** [comment]
Main fix done, still considering whether more failures should be logged on bridge server side (probably?).

File: bridge/src/main/java/io/stargate/bridge/service/ExceptionHandler.java [hunk 1]
```diff
@@ -43,9 +43,9 @@ public class ExceptionHandler {
   static final Metadata.Key<QueryOuterClass.CasWriteUnknown> CAS_WRITE_UNKNOWN_KEY =
       ProtoUtils.keyForProto(QueryOuterClass.CasWriteUnknown.getDefaultInstance());
 
-  private final StreamObserver<QueryOuterClass.Response> responseObserver;
+  private final StreamObserver<?> responseObserver;
```
Comment (tatu-at-datastax) [inline bridge/src/main/java/io/stargate/bridge/service/ExceptionHandler.java:46]: Parameterization not used for anything. Could have changed it to something like `GeneratedMessageV3` (common base class for protobuf-generated message classes) but does not seem valuable.

**Comment by tatu-at-datastax** [comment]
@ivansenic Aside from main changes didn't yet add more logging of failures: feel free to change if you want, as we need this to be merged for build. Or I can change it when I get online. And we can always change it later on either way.

File: bridge/src/test/java/io/stargate/bridge/service/SchemaOperationsTest.java [hunk 2]
```diff
@@ -253,4 +256,30 @@ public void schemaTableWithMaterializedView() {
...
+
+  @Test
+  @DisplayName("Handle and propagate server-side NPE appropriately")
+  public void schemaTableWithServerSideNPE() {
+    // Given
+    StargateBridgeGrpc.StargateBridgeBlockingStub stub = makeBlockingStub();
+    when(persistence.decorateKeyspaceName(any(String.class), any())).thenReturn("my_stuff");
+
+    // Only need minimal schema for bootstrapping
+    io.stargate.db.schema.Schema schema = io.stargate.db.schema.Schema.build().build();
```
Comment (ivansenic) [inline bridge/src/test/java/io/stargate/bridge/service/SchemaOperationsTest.java:268]: Codacy complains here and I guess is correct this time:

> io.stargate.db.schema.Schema schema = io.stargate.db.schema.Schema.build().build();

File: bridge/src/main/java/io/stargate/bridge/service/ExceptionHandler.java (hunk 1 above)
Comment (ivansenic) [reply bridge/src/main/java/io/stargate/bridge/service/ExceptionHandler.java:46]: Leave it.. seems enough for exception handling

**Review by ivansenic — CHANGES_REQUESTED** [review]
We are missing the protection in the `io.stargate.bridge.service.BridgeService#executeQueryWithSchema`, exactly same issues can occur there if for example runtime exception occurs with persistence interaction, calling io.stargate.bridge.service.SchemaHandler#buildKeyspaceDescription, etc

Furthermore, the code in current state is complete spaghetti code. There is absolutely no rules on which components do what, separation of concerns, etc.. Let's solve this quickly and then let's have a ticket to rewrite this, including tests..

**Review by ivansenic — APPROVED** [review]
Approving, @tatu-at-datastax will submit another PR for other fixes..

File: bridge/src/test/java/io/stargate/bridge/service/SchemaOperationsTest.java (hunk 2 above)
Comment (tatu-at-datastax) [reply bridge/src/test/java/io/stargate/bridge/service/SchemaOperationsTest.java:268]: > Codacy complains here and I guess is correct this time:
> 
> > io.stargate.db.schema.Schema schema = io.stargate.db.schema.Schema.build().build();

Ah. Yes, will remove. Copy-paste flag. For once Codacy was right :)

**Pull request merged**